support/cas-server-support-discovery-profile/src/main/java/org/apereo/cas/discovery/CasServerProfileRegistrar.java

```diff
@@ -122,6 +124,25 @@ private Object locateSubtypesByReflection(final Function<Class, Object> mapper,
         return clients.findAllClients().stream().map(Client::getName).collect(Collectors.toSet());
     }
 
+    private Set<String> locateAvailableAttributes() {
```

Comment:
This is perfectly good, but I think it might be slightly better if you simply inject the `Map` into this class, rather than building here every time. I can think of a number of advantages with that approach specifically wrt to reloadability of the settings.

Reply:
Sorry to be dense, but I am not following where the injection would come from.

Reply:
No worries. The injection would come from whatever class that is building the bean instance for `CasServerProfileRegistrar`. It's probably the related @Configuration class. You can build the map fully there, and you also wouldn't have to worry about null checks on `attributeRepository`. The suspected candidate would be `CasDiscoveryProfileConfiguration.java`.

Then the final Map/Set gets injected into the class here (and you can, if you are brave, mark the bean with RefreshScope)